test/test_valgrind.sh

```diff
@@ -117,6 +117,9 @@ for test in $(ls -1 umf_test-*); do
 	umf_test-memspace_lowest_latency)
 		FILTER='--gtest_filter="-*allocLocalMt*"'
 		;;
+	umf_test-memoryPool)
+		FILTER='--gtest_filter="-*allocMaxSize*"'
```

Comment:
Why we are disabling valgrind here - we are expecting fail of the allocation, so valgrind should be fine.

Reply:
Memcheck fails the test with SIZE_MAX allocation: https://github.com/PatKamin/unified-memory-framework/actions/runs/11481177678/job/31952726455#step:6:116